---
layout: post
title: Sun. Oct 28, 2018
---

## Methylation analysis:  P. generosa juvenile OA response  
### bismark currently running on full data set  
### mapping and coverage analysis with subsetted samples (10K reads each)  
	
[bismark summary](https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/analyses/BismarkAlignments/bismark_summary_report.html)  
		- all samples have > 50% mapping efficiency with Bismark settings --score_min L,0,-1.2 -I 60 --non_directional   
		- deduplication removed very few reads, per usual   
		- % methylation is on average 20%  
	
[methylkit summary](https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/analyses/MethylKit_JuviPgenr_diffpH/GeoduckSubsetAlignmentTest.md)
		- coverage plots don't appear to obviously show any particular sample or condition should be excluded due to low coverage.this may become more obvious in the full analysis or downstream analysis  
		- the genome region > 60000 seems to have low coverage in all samples    
		[plot of genome regions with no coverage organized by length of treatment](https://raw.githubusercontent.com/shellywanamaker/Shelly_Pgenerosa/master/analyses/MethylKit_JuviPgenr_diffpH/GeoduckSubsetAlignmentTest_files/figure-markdown_github/unnamed-chunk-9-14.png)







## C. virginica
### bismark run completed on mox
	- last part failed because I copied from jupyter notebook and had '!' in the code still
	- need to edit this part for future scripts, but for now i will complete the analysis on ostrich in my jupyter notebook
		- generate summary file
		- generate sorted bam files for methylkit analysis
	- need to rsync everything onto Metacarcinus from Mox gscratch

#### failed code
		/var/spool/slurm/d/job394229/slurm_script: line 47: !cat: command not found
		/var/spool/slurm/d/job394229/slurm_script: line 51: !sed: command not found
		/var/spool/slurm/d/job394229/slurm_script: line 56: fg: no job control
		xargs: samtools: No such file or directory
		/var/spool/slurm/d/job394229/slurm_script: line 61: -o: command not found

#### code excerpt
		46 #create summary report
		47 !cat /gscratch/srlab/strigg/analyses/20181004/*PE_report.txt | \
		48 grep 'Mapping\ efficiency\:' | \
		49 cat - /gscratch/srlab/strigg/analyses/20181004/*.deduplication_report.txt > /gscratch/srlab/strigg/analyses/20181004/zr2096_mapping_dedup_summary.txt
		50 #clean up summary report
		51 !sed 's/Mapping\ efficiency\://g' /gscratch/srlab/strigg/analyses/20181004/zr2096_mapping_dedup_summary.txt | \
		52 sed 's/Total\ number\ duplicated\ alignments\ removed\://g' | \
		53 sed 's/ //g' | awk '{print $1}' > /gscratch/srlab/strigg/analyses/20181004/zr2096_mapping_dedup_summary_clean.txt
		54 
		55 #sort bams
		56 %%bash
		57 find /gscratch/srlab/strigg/analyses/20181004/*deduplicated.bam| \
		58 xargs basename -s _s1_R1_bismark_bt2_pe.deduplicated.bam | \
		59 xargs -I{} samtools \
		60 sort /gscratch/srlab/strigg/analyses/20181004/{}_s1_R1_bismark_bt2_pe.deduplicated.bam \ 
		61 -o /gscratch/srlab/strigg/analyses/20181004/{}_dedup.sorted.bam



