---
layout: post
title: Run nf-core RNAseq and Methylseq pipelines on _P. tuahiniensis_ data
author: Shelly Wanamaker
tags: RNA-seq methylseq nf-core coral E5
---
#Purpose

### This post is related to github issue [# 2174](https://github.com/RobertsLab/resources/issues/2174)

The purpose of this entry is to align RNAseq and EM-seq data to the _P. tuahiniensis_ genome using nf-core pipelines on Klone.

# Methods
### Copy genome files to klone
```
# show path
pwd
/gscratch/srlab/strigg/GENOMES

# copy genome
wget https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_meandrina_HIv1.assembly.fasta

#copy gtf file
wget https://github.com/urol-e5/timeseries_molecular/raw/d5f546705e3df40558eeaa5c18b122c79d2f4453/F-Ptua/data/Pocillopora_meandrina_HIv1.genes-validated.gtf

# copy gff file
wget https://github.com/urol-e5/timeseries_molecular/raw/d5f546705e3df40558eeaa5c18b122c79d2f4453/F-Ptua/data/Pocillopora_meandrina_HIv1.genes-validated.gff3
```
Note: I had to remove the '3' from the file extention of the gff and zip it to run the nf-core RNAseq pipeline


### Copy RNAseq data to Klone
```
# open screen session (reopened existing session)
screen -r RNAseq

# start interactive node
salloc -A srlab -p cpu-g2-mem2x -N 1 -c 1 --mem=16GB --time=16:00:00

# copy data
rsync --progress --verbose --archive shellytrigg@gannet.fish.washington.edu:/volume2/web/gitrepos/urol-e5/timeseries_molecular/F-Ptua/output/01.00-F-Ptua-RNAseq-trimming-fastp-FastQC-MultiQC/*.gz /gscratch/scrubbed/strigg/analyses/20250421_RNAseq
```
### Copy WGBS to klone

```
# open screen session
screen -r methylseq

# start interactive node
salloc -A srlab -p cpu-g2-mem2x -N 1 -c 1 --mem=16GB --time=16:00:00

# copy data 
rsync --progress --verbose --archive shellytrigg@gannet.fish.washington.edu:/volume2/web/gitrepos/urol-e5/timeseries_molecular/F-Ptua/output/01.00-F-Ptua-WGBS-trimming-fastp-FastQC-MultiQC/*.gz /gscratch/scrubbed/strigg/analyses/20250421_methylseq

```
### Run RNAseq pipeline
```
# open screen session 
screen -r RNAseq

# start interactive node
salloc -A srlab -p cpu-g2-mem2x -N 1 -c 1 --mem=16GB --time=24:00:00

# activate conda environment
mamba activate nextflow

# run pipeline
nextflow run nf-core/rnaseq -resume \
-c /gscratch/srlab/nextflow/uw_hyak_srlab.config \
--input /gscratch/scrubbed/strigg/analyses/20250421_RNAseq/samplesheet.csv \
--outdir /gscratch/scrubbed/strigg/analyses/20250421_RNAseq \
--gtf /gscratch/srlab/strigg/GENOMES/Pocillopora_meandrina_HIv1.genes-validated.gtf \
--gff /gscratch/srlab/strigg/GENOMES/Pocillopora_meandrina_HIv1.genes-validated.gff.gz \
--fasta /gscratch/srlab/strigg/GENOMES/Pocillopora_meandrina_HIv1.assembly.fasta \
--skip_trimming \
--aligner star_salmon \
--skip_pseudo_alignment \
--multiqc_title Pmeandrina_RNAseq \
--deseq2_vst
```

### Run methylseq pipeline
```
# open screen session 
screen -r methylseq

# start interactive node
salloc -A srlab -p cpu-g2-mem2x -N 1 -c 1 --mem=16GB --time=72:00:00

#activate conda environment
mamba activate nextflow

#run methylseq pipeline

nextflow run nf-core/methylseq \
-c /gscratch/srlab/strigg/bin/uw_hyak_srlab.config \
--input /gscratch/scrubbed/strigg/analyses/20250421_methylseq/samplesheet.csv \
--outdir /gscratch/scrubbed/strigg/analyses/20250421_methylseq \
--fasta /gscratch/srlab/strigg/GENOMES/Pocillopora_meandrina_HIv1.assembly.fasta \
--em_seq
-resume \
-with-report nf_report.html \
-with-trace \
-with-timeline nf_timeline.html \
--skip_trimming \
--nomeseq 

```

